src/OpenTelemetry.Exporter.Geneva/Internal/EventHeader/EventHeaderLogExporter.cs

```diff
@@ -0,0 +1,352 @@
+// Copyright The OpenTelemetry Authors
+// SPDX-License-Identifier: Apache-2.0
+
+#if NET
+
+using System.Globalization;
+using Microsoft.LinuxTracepoints.Provider;
+using OpenTelemetry.Exporter.Geneva.Tld;
+using OpenTelemetry.Exporter.Geneva.Transports;
+using OpenTelemetry.Internal;
+using OpenTelemetry.Logs;
+
+namespace OpenTelemetry.Exporter.Geneva.EventHeader;
+
+internal class EventHeaderLogExporter : TldLogCommon, IDisposable
+{
+    internal const int StringLengthLimit = (1 << 14) - 1; // 16 * 1024 - 1 = 16383
+
+    private static readonly ThreadLocal<EventHeaderDynamicBuilder> EventBuilder = new();
+    private static readonly Action<LogRecordScope, EventHeaderLogExporter> ProcessScopeForIndividualColumnsAction = OnProcessScopeForIndividualColumns;
+
+    private readonly ValueTuple<byte[], byte[]>? repeatedPartAFields;
+
+#pragma warning disable CA2213 // Disposable fields should be disposed: logsTracepoint is a local reference set in the ctor. It's lifecycle is managed by UnixUserEventsDataTransport and is disposed in EventHeaderDynamicProvider.Dispose
+    private readonly EventHeaderDynamicTracepoint logsTracepoint;
+#pragma warning restore CA2213 // Disposable fields should be disposed
+
+    public EventHeaderLogExporter(GenevaExporterOptions options)
+        : base(options)
+    {
+        Guard.ThrowIfNull(options);
+
+        this.logsTracepoint = UnixUserEventsDataTransport.Instance.RegisterUserEventProviderForLogs();
+
+        if (options.PrepopulatedFields != null)
+        {
+            var eb = EventBuilder.Value ??= new EventHeaderDynamicBuilder();
+
+            eb.Reset("_"); // EventName does not matter here as we only need the serialized key-value pairs
+
+            foreach (var entry in options.PrepopulatedFields)
+            {
+                var key = entry.Key;
+                var value = entry.Value;
+
+                if (entry.Key == Schema.V40.PartA.Ver)
+                {
+                    continue;
+                }
+
+                TldExporter.V40_PART_A_TLD_MAPPING.TryGetValue(key, out var replacementKey);
+                var keyToSerialize = replacementKey ?? key;
+                EventHeaderExporter.Serialize(eb, keyToSerialize, value);
+            }
+
+            this.repeatedPartAFields = eb.GetRawFields();
```

Comment:
In TldExporter this line is within the foreach, should that also be moved outside?